packages/select/src/select.vue

```diff
@@ -723,7 +731,6 @@
             value.push(option.value);
           }
           this.$emit('input', value);
-          this.emitChange(value);
```

Comment:
You are right, that is how native selects act. 

Even though Element doesn't allow collapsed cycling of options we've changed that in our cognito fork though. It should still work (if I remember correctly) because the change event is only not fired if the menu is open.